lib/widgets/message_list.dart

```diff
@@ -781,58 +783,81 @@ class MessageWithPossibleSender extends StatelessWidget {
+                            fontSize: 18,
+                            height: (22 / 18),
+                          ).merge(weightVariableTextStyle(context, wght: 600,
+                              wghtIfPlatformRequestsBold: 900)),
+                          overflow: TextOverflow.ellipsis),
+                      ),
+                    ]),
+                )),
+            ),
+            Text(time,
+              style: TextStyle(
+                color: _kMessageTimestampColor,
+                fontFamily: 'Source Sans 3',
+                fontSize: 16,
+                height: (18 / 16),
+                fontFeatures: const [FontFeature.enable('c2sc'), FontFeature.enable('smcp')],
+              ).merge(weightVariableTextStyle(context))),
+          ]));
+    } else {
+      senderWidget = const SizedBox.shrink();
+    }
+
     return GestureDetector(
       behavior: HitTestBehavior.translucent,
       onLongPress: () => showMessageActionSheet(context: context, message: message),
-      // TODO clean up this layout, by less precisely imitating web
-      child: Padding(
-        padding: const EdgeInsets.only(top: 2, bottom: 3, left: 8, right: 8),
-        child: Row(crossAxisAlignment: CrossAxisAlignment.start, children: [
-          item.showSender
-            ? Padding(
-                padding: const EdgeInsets.fromLTRB(3, 6, 11, 0),
-                child: GestureDetector(
-                  onTap: () => Navigator.push(context,
-                    ProfilePage.buildRoute(context: context,
-                      userId: message.senderId)),
-                  child: Avatar(size: 35, borderRadius: 4,
-                    userId: message.senderId)))
-            : const SizedBox(width: 3 + 35 + 11),
-          Expanded(
-            child: Column(
-              crossAxisAlignment: CrossAxisAlignment.stretch,
-              children: [
-                if (item.showSender) ...[
-                  const SizedBox(height: 3),
-                  GestureDetector(
-                    onTap: () => Navigator.push(context,
-                      ProfilePage.buildRoute(context: context,
-                        userId: message.senderId)),
-                    child: Text(message.senderFullName, // TODO get from user data
-                      style: const TextStyle(fontWeight: FontWeight.bold))),
-                  const SizedBox(height: 4),
-                ],
-                MessageContent(message: message, content: item.content),
-                if ((message.reactions?.total ?? 0) > 0)
-                  ReactionChipsList(messageId: message.id, reactions: message.reactions!)
-              ])),
-          Container(
-            width: 80,
-            padding: const EdgeInsets.only(top: 4, right: 16 - 8),
-            alignment: Alignment.topRight,
-            child: Text(time, style: _kMessageTimestampStyle)),
-        ])));
+      child: Column(
+        children: [
+          senderWidget,
```

Comment:
nit:
```suggestion
          if (senderWidget != null) senderWidget,
```

and then instead of `SizedBox.shrink`, can set the variable to null. I think that'd make it a bit more evident when looking at this spot that it's conditional whether there's really any sender anything being shown.